### PR TITLE
Counter underflow fixes

### PR DESCRIPTION
**Pull request by therandomstring**
Hopes to fix certain counter over/underflow issues, likely caused by narrowing type conversions.

### CODE REVIEW

File: lib/logsource.h [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 typedef struct _LogSourceOptions
 {
-  gint init_window_size;
+  gsize init_window_size;
```
Comment (MrAnno) [inline lib/logsource.h:36]: This is a `-1`-initialized value, let's use gssize.

(I know, `initial_window_size` is unsigned and we'd use half of the representable window sizes this way, but per-source windows should not be that high.)

**Review by MrAnno — CHANGES_REQUESTED** [review]
Just a minor thing about `init_window_size`

File: lib/logsource.h (hunk 1 above)
Comment (therandomstring) [reply lib/logsource.h]: Thank you for the catch. Fixed

Comment (MrAnno) [reply lib/logsource.h]: Thanks! :)

**Review by MrAnno — APPROVED** [review]
LGTM.

**Pull request merged**